File: code/java/src/main/java/composite/Playlist.java
package composite;

import java.util.ArrayList;
import java.util.List;

class Playlist implements PlaylistComponent {
    private List<PlaylistComponent> components;

    Playlist() {
        this.components = new ArrayList<>();
    }

    void addComponent(PlaylistComponent component) {
        components.add(component);
    }

    void removeComponent(PlaylistComponent component) {
        components.remove(component);
    }

    @Override
    public void playSong() {
        // Perform playlist-specific operation, such as playing all songs
        System.out.println("Playing playlist:");
        for (PlaylistComponent component : components) {
            component.playSong();
        }
    }
}

File: code/python/composite/Playlist.py
from PlaylistComponent import PlaylistComponent


class Playlist(PlaylistComponent):
    def __init__(self):
        self.components = []

    def add_component(self, component):
        self.components.append(component)

    def remove_component(self, component):
        self.components.remove(component)

    def play_song(self):
        print("Playing playlist:")
        for component in self.components:
            component.play_song()
